vividus-plugin-mobitru/src/main/java/org/vividus/mobitru/client/MobitruFacadeImpl.java

```diff
@@ -66,6 +67,20 @@ public MobitruFacadeImpl(MobitruClient mobitruClient)
     @Override
     public String takeDevice(DesiredCapabilities desiredCapabilities) throws MobitruOperationException
     {
+        Map<String, Object> capabilities = desiredCapabilities.asMap();
+        //use different API in case if udid is provided in capabilities
+        //it's required in some cases like if the device is already taken
+        if (!isSearchForDevice(desiredCapabilities)
+                &&
+                (capabilities.containsKey(APPIUM_UDID) || capabilities.containsKey(UDID)))
+        {
+            Object deviceIdObj = capabilities.getOrDefault(APPIUM_UDID, capabilities.get(UDID));
+            String udid = Optional.ofNullable(deviceIdObj).
+                    map(String::valueOf).
+                    orElseThrow();
```

Comment:
afaik DesiredCapabilities drops all props with null values so its guaranteed to not be null

Reply:
done